smart_card_connector_app/src/application_unittest.cc

```diff
@@ -1208,6 +1226,47 @@ TEST_F(SmartCardConnectorApplicationSingleClientTest,
             SCARD_S_SUCCESS);
 }
 
+// `SCardReconnect()` call from JS succeeds when using the same parameters as
+// the previous `SCardConnect()` call.
+TEST_F(SmartCardConnectorApplicationSingleClientTest, SCardReconnect) {
+  // Arrange:
+  TestingSmartCardSimulation::Device device;
+  device.id = 123;
+  device.type = TestingSmartCardSimulation::DeviceType::kGemaltoPcTwinReader;
+  device.card_type = TestingSmartCardSimulation::CardType::kCosmoId70;
+  SetUsbDevices({device});
+  StartApplication();
+  SetUpJsClient();
+  SetUpSCardContext();
+  // Connect to the card.
+  SCARDHANDLE scard_handle = 0;
+  DWORD active_protocol = 0;
+  EXPECT_EQ(SimulateConnectCallFromJsClient(
+                kFakeHandlerId, scard_context(), kGemaltoPcTwinReaderPcscName0,
+                SCARD_SHARE_SHARED, SCARD_PROTOCOL_ANY, scard_handle,
+                active_protocol),
+            SCARD_S_SUCCESS);
+  EXPECT_NE(scard_handle, 0);
+  EXPECT_EQ(active_protocol, static_cast<DWORD>(SCARD_PROTOCOL_T1));
+
+  // Act:
+  // Reconnect using the same parameters.
```

Comment:
mini-nit: I guess it's not exactly the same, SCARD_LEAVE_CARD is new

Reply:
Fair point; rephrased.